### PR TITLE
Leapfrog fix

### PR DESCRIPTION
**Pull request by brryan**
## PR Summary

The `particle_leapfrog_outflow` test is currently breaking CI intermittently.

I think the issue is that, for zero particles on a rank, we don't initialize all `send_size`s to `0` so sometimes they're full of garbage. This PR seems to cycle correctly.

## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] Docs build
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Review by Yurlungur — APPROVED** [review]

**Comment by brryan** [comment]
OK there's an additional issue which is that our `sort()` command is currently disabled for HIP. I'm going to see if I can't fix this

**Comment by pgrete** [comment]
> OK there's an additional issue which is that our `sort()` command is currently disabled for HIP. I'm going to see if I can't fix this

This should be fixed with Kokkos 4.4 which has support for sorting on hip (though the interface would need to be updated).

File: .github/workflows/ci-short.yml [hunk 1]
```diff
@@ -154,17 +154,19 @@ jobs:
...
+      #    cmake --build build -t advection-example
+      #    cd build
+      #    ctest -R regression_mpi_test:output_hdf5
       # Test example with swarms
       - name: particle-leapfrog
         run: |
           cmake --build build -t particle-leapfrog
           cd build
-          ctest -R regression_mpi_test:particle_leapfrog
+          /usr/bin/mpiexec -n 2 --allow-run-as-root /__w/parthenon/parthenon/build/example/particle_leapfrog/particle-leapfrog -i /__w/parthenon/parthenon/tst/regression/test_suites/particle_leapfrog/parthinput.particle_leapfrog parthenon/job/problem_id=gold --kokkos-map-device-id-by=mpi_rank
```
Comment (alexrlongne) [inline .github/workflows/ci-short.yml]: Does this fix the behavior that allowed the CI to pass even when this failed?

File: src/interface/swarm_comms.cpp [hunk 2]
```diff
@@ -269,6 +269,11 @@ void Swarm::LoadBuffers_() {
 
     // Remove particles that were loaded to send to another block from this block
     RemoveMarkedParticles();
+  } else {
+    for (int n = 0; n < pmb->neighbors.size(); n++) {
+      const int bufid = pmb->neighbors[n].bufid;
+      vbswarm->send_size[bufid] = 0;
```
Comment (alexrlongne) [inline src/interface/swarm_comms.cpp:275]: Thanks, sorry I missed this.

**Comment by brryan** [comment]
> > OK there's an additional issue which is that our `sort()` command is currently disabled for HIP. I'm going to see if I can't fix this
> 
> This should be fixed with Kokkos 4.4 which has support for sorting on hip (though the interface would need to be updated).

Yes agreed, I can make a separate WIP PR to switch to `Kokkos::sort()` after we upgrade parthenon to Kokkos 4.4

File: .github/workflows/ci-short.yml (hunk 1 above)
Comment (brryan) [reply .github/workflows/ci-short.yml]: This was actually me doing experiments trying to catch why the CI was still failing (the `sort()` issue); I reverted this change.

**Review by pdmullen — APPROVED** [review]

**Pull request merged**